File: mixdrop-back/mixdrop-back/Models/Entities/Battle.cs
namespace mixdrop_back.Models.Entities;

public class Battle
{
    public int Id { get; set; }
    public bool IsAgainstBot { get; set; } = false;
    public BattleState BattleState { get; set; }
    public int BattleStateId { get; set; }
    public ICollection<UserBattle> BattleUsers { get; set; } = new List<UserBattle>();
    public DateTime CreatedAt { get; set; }
    public DateTime FinishedAt { get; set; } = DateTime.UtcNow.AddMonths(1); // Valor por defecto
}

File: mixdrop-front/src/app/models/battle.ts
import { User } from "./user";
import { UserBattle } from "./user-battle";

export interface Battle {
    id: number,
    accepted : boolean,
    isAgainstBot : boolean,
    isPlaying : boolean,
    battleUsers : UserBattle[],
    user : User,
    createdAt: Date,
    finishedAt: Date
}
